5 westport fixes name, indexing, offshore chunking

rosepearson opened the pull request:
This pull request is in response to several issues encountered when producing a DEM for Westport.
Changes include:
1. Name the created dem with the binning approach -> idw
2. Ensure the created dem has positively increasing x and y indexing
3. Apply a tiling approach to the offshore interpolation to limit the maximum requirement

For this PR a code review of changes is probably all that is required.

LukeParky reviewed (changes requested):
Overall high quality code. Some of this is pushing me in domain knowledge now but I added some suggestions for simplifying code

rosepearson commented on src/geofabrics/dem.py:
```diff
@@ -164,9 +168,11 @@ def _set_up(self):
+        # check if the raster origin has been moved by PDAL writers.gdal
+        raster_origin = [dem_temp.x.data.min() - self.catchment_geometry.resolution/2,
+                         dem_temp.y.data.min() - self.catchment_geometry.resolution/2]
+        if self.raster_origin[0] != raster_origin[0] or self.raster_origin[1] != raster_origin[1]:
```
Does that work for lists! I will try this now.

rosepearson replied:
Done - nice suggestion

rosepearson commented on src/geofabrics/dem.py:
```diff
@@ -242,8 +248,25 @@ def dem(self):
+            y = y[::-1]
+        dem = dem.reindex(x=x, y=y)
+        return dem
+
```
Done. Nice suggestion.

rosepearson commented on src/geofabrics/processor.py:
```diff
@@ -128,7 +133,8 @@ def run(self):
         if (self.catchment_geometry.land_and_foreshore_without_lidar(self.catchment_lidar.extents).geometry.area.sum()
-                > self.catchment_geometry.land_and_foreshore.area.sum() * area_threshold):
+                > self.catchment_geometry.land_and_foreshore.area.sum() * area_threshold and
+                self.check_instruction_path('reference_dems')):
```
Ok - I've had a good. Reopen if still needs work!

rosepearson commented:
> Overall high quality code. Some of this is pushing me in domain knowledge now but I added some suggestions for simplifying code

Thanks @LukeParky. Great suggestions as always. I have addressed them. Let me know what you think!

LukeParky commented:
Thanks for those changes, I approve it, might be worth keeping it open for @AliceHarang to double check

rosepearson commented:
> Thanks for those changes, I approve it, might be worth keeping it open for @AliceHarang to double check

Whoops - probably would have been. I was a bit keen.
